### ISSUE TITLE
Module settings view shows errors even after successfully saving settings

### ISSUE THREAD

**Issue by eugene-manuilov** [google/site-kit-wp#3569]
## Bug Description

When I successfully submitted settings, the settings preview screen keeps showing errors that had been shown during editing settings. We should reset errors for a module when settings have been saved or changes have been canceled.

## Steps to reproduce

1. Go to the Google Analytics console and create a new account.
1. Activate the Analytics module and connect the newly created account.
1. Go to the settings page and make sure you see your selection on the settings preview screen.
1. Go to the Google Analytics console again and delete the newly created account.
1. Go back to the settings page and click on the "Edit" settings button.
1. When the settings edit form loads, you should see errors on the form because it can't find the deleted account (see screenshot below)
1. Select another account, property, and profile and submit changes.
1. Now you should see the settings preview screen, but errors from the settings edit form don't disappear.

## Screenshots

![sitekit 10uplabs com_wp-admin_admin php_page=googlesitekit-settings](https://user-images.githubusercontent.com/873049/121932952-1ea1de80-cd4e-11eb-82e3-c9ca3711a340.png)

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* Module store errors should be cleared after successfully saving settings
* Secondarily, module stores should not show duplicate error messages 

### Prevent duplicate error messages by filtering getErrors calls

* Filter errors with the exact same error message from the [`StoreErrorNotices` component](https://github.com/google/site-kit-wp/blob/7ee265439d32c19415370d75eeca72f6505e252d/assets/js/components/StoreErrorNotices.js#L46) by adding:
```js
const existingErrorMessages = [];

	return errors
		.filter( ( error ) => {
			if (
				! error?.message ||
				existingErrorMessages.includes( error.message )
			) {
				return false;
			}

			existingErrorMessages.push( error.message );

			return true;
		} )
		// Include other map methods in original file here…
```
* Clear all `modules/analytics` errors when selecting a new account. (See: 407e36905d4b21e80eb552e45ec32195857f9851) as well as when the settings screen is closed or the new settings are submitted.

### Test Coverage

* N/A

### Visual Regression Changes

* N/A

## QA Brief

1. Go to the Google Analytics console and create a new account.
2. Activate the Analytics module and connect the newly created account.
3. Go to the settings page and make sure you see your selection on the settings preview screen.
4. Go to the Google Analytics console again and delete the newly created account.
5. Go back to the settings page and click on the "Edit" settings button.
6. When the settings edit form loads, you should see an error on the form because it can't find the deleted account. You shoud only see 
1. Select another account, property, and profile and submit changes.
1. Now you should see the settings preview screen, but errors from the settings edit form don't disappear.

## Changelog entry

* Don't show duplicate errors when loading deleted Analytics accounts in the settings.

**Comment by aaemnnosttv** [comment]
This happens on the edit view as well. The duplicate error message is odd too.

**Comment by eugene-manuilov** [comment]
> ... The duplicate error message is odd too.

The duplicate error message caused by two failing requests to different endpoints.

**Comment by aaemnnosttv** [comment]
@felixarntz anything to change or add about the ACs here?

**Comment by felixarntz** [comment]
@aaemnnosttv LGTM!

**Comment by eugene-manuilov** [comment]
> Filter errors with the exact same error message from the getErrors selector by adding:

@tofumatt let's better not change the results of the `getErrors` selector and filter out duplicate errors in the `StoreErrorNotices` component.

> Clear all modules/analytics errors when selecting a new account.

We need to clear all module errors when we either close settings or before submitting the settings form.

**Comment by eugene-manuilov** [comment]
> Clear all modules/analytics errors when selecting a new account.

@tofumatt I think this is redundant, we don't need to clear all errors when we select a different account because fetch requests will clear their errors themselves. Let's clear all errors only when we close the settings form or just before submitting the form.

**Comment by eugene-manuilov** [comment]
@tofumatt ah... yes, I think you are right. Okey, then looks good to me.

**Comment by eugene-manuilov** [comment]
IB :heavy_check_mark:

**Comment by wpdarren** [comment]
@kuasha420 this is the screencast.   Just to add, after finishing this recording, I clicked on the 'Confirm changes' button. This then took me back to the settings overview for Analytics.  I clicked on edit again and the same red border appeared. It wasn't until I went away from settings (e.g. dashboard) and back in again to settings that it disappeared.  

It's quite possible that this is outside of the ticket and I am happy to create a new one for this. 

https://user-images.githubusercontent.com/73545194/137159145-a890279e-fc9b-4050-939f-5facafaaddb4.mp4
